File: DesignPatterns/src/C++/002-GOFPatterns/003-BehavioralPatterns/006-Observer/IntegerObserversWithOldValue/HexObserver.cpp
#include <iostream>
#include <cstdio>
#include "HexObserver.h"

using namespace std;

void HexObserver::Update(int oldVal)
{
    cout << std::hex << "Old Value:" << oldVal << '\n';
    cout << std::hex << "Old Value:" << m_pi->GetVal() << '\n';
}

File: src/DesignPatterns/src/Java/002-GOF-Patterns/003-BehavioralPatterns/006-Observer/Observer/src/org/csystem/app/HexObserver.java
package org.csystem.app;

public class HexObserver implements Observer<IntValue> {
    @Override
    public void update(IntValue oldVal, IntValue newVal)
    {
        System.out.printf("Old Value:%X%n", oldVal.getValue());
        System.out.printf("New Value:%X%n", newVal.getValue());
    }
}
